torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -206,23 +217,83 @@ static PyObject* get_type_of_overloaded_arg(PyObject* obj_or_type) {
 }
 
 // See Note: [Overloaded args] for what they hold
-auto handle_torch_function_no_python_arg_parser(at::ArrayRef<py::handle> overloaded_args, PyObject* args, PyObject* kwargs, const char* func_name, PyObject* torch_api_function, const char* module_name, const char* torch_function_name) -> PyObject* {
+auto handle_torch_function_no_python_arg_parser(
+    at::ArrayRef<py::handle> overloaded_args,
+    PyObject* args,
+    PyObject* kwargs,
+    const char* func_name,
+    PyObject* torch_api_function,
+    const char* module_name,
+    TorchFunctionName torch_function_name) -> PyObject* {
+  const char* torch_function_name_str = nullptr;
+  switch (torch_function_name) {
+    case TorchFunctionName::TorchFunction:
+      torch_function_name_str = "__torch_function__";
```

Comment:
Nope. torch_function_mode_name was when I was thinking about changing the function name for TorchDispatchMode objects. We're not allowed to change the name of the torch function on Tensor-like objects though. (Do you think TorchDispatchMode should define `__torch_function__`, or something differently named?)

Reply:
> But we will never change it, so I guess it is not really important to have it defined only once.

Yeah, we're not going to change it. Hard code away.

However, we *do* have a choice before landing this PR, which is whether or not to have the mode objects define `__torch_function__`. What is weird about reusing the name in this way is I can also pass a mode object AS AN ARGUMENT to a torch API function, and it will trigger the alternate codepath and stick you in the `__torch_function__` which will promptly fail to do anything reasonable because it was expecting to be used as a mode, and not expecting to see a mode object as one of the arguments.

The reason I didn't change this name for the PR was that it seemed unlikely real users would make this mistake, and I really didn't want to have to explain *yet another* magic method name to users of this feature.